vm_network_migration/migrations/instance_group_network_migration.py

```diff
@@ -0,0 +1,270 @@
+
+    def network_migration(self, instance_group_name,
+                          network_name,
+                          subnetwork_name, preserve_external_ip):
+        """ The main method of the instance network migration process
+
+        Args:
+            network_name: target network name
+            subnetwork_name: target subnetwork name
+            preserve_external_ip: preserve the external IP of the instances
+            in an unmanaged instance group
+
+        Returns: None
+
+        """
+        if self.instance_group == None:
+            self.instance_group = self.build_instance_group(instance_group_name)
+        if isinstance(self.instance_group, UnmanagedInstanceGroup):
+            try:
+                self.migrate_unmanaged_instance_group(network_name,
+                                                      subnetwork_name,
+                                                      preserve_external_ip)
+            except Exception as e:
+                warn(e, Warning)
+                print(
+                    'The migration is failed. Rolling back to the original instance group.')
+                self.rollback_unmanaged_instance_group()
+
+        else:
+            try:
+                self.migrate_managed_instance_group(network_name,
+                                                    subnetwork_name)
+            except Exception as e:
+                warn(e, Warning)
+                print(
+                    'The migration is failed. Rolling back to the original instance group.')
+                self.rollback_managed_instance_group()
+
+    def migrate_unmanaged_instance_group(self, network_name, subnetwork_name,
+                                         preserve_external_ip):
+        """ Migrate the network of an unmanaged instance group.
+        The instances belonging to this instance group will
+        be migrated one by one.
+
+        Args:
+            network_name: target network
+            subnetwork_name: target subnetwork
+            preserve_external_ip: whether preserving the external IP
+            of the instances in the group
+
+        """
+        if self.region == None:
+            self.region = self.get_region()
+        instance_network_migration = InstanceNetworkMigration(self.project,
+                                                              self.zone)
+        print(
+            "Migrating all the instances in the instance group to the new network.")
+        for instance in self.instance_group.instances:
+            instance_network_migration.instance = instance
+            instance_network_migration.network_migration(instance.name,
+                                                         network_name,
+                                                         subnetwork_name,
+                                                         preserve_external_ip)
+        print("Modifying the instance group's configs with the new network.")
```

Comment:
"Modifying the instance group config to use the new network."

Reply:
Addressed.